app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.java

```diff
@@ -326,8 +321,11 @@ private void viewDay(boolean reload) {
             int roomIndex = MyApp.roomList.get(i);
             fillRoom(roomView, roomIndex, MyApp.lectureList, boxHeight);
         }
-        scrollToCurrent(mDay, boxHeight);
+        if (MainActivity.getInstance().getScrollToCurrentTimeSlot()) {
+            scrollToCurrent(mDay, boxHeight);
+        }
         updateNavigationMenuSelection();
+        MainActivity.getInstance().resetScrollToCurrentTimeSlot();
```

Comment:
The following code allows to reduce the public interface of `MainActivity` for this use case to a **single** method. The reset is done internally:

``` java
MainActivity.getInstance().shouldScheduleScrollToCurrentTimeSlot(() -> {
    scrollToCurrent(mDay, boxHeight);
    return null;
});
```

in MainActivity.java:

``` java
public void shouldScheduleScrollToCurrentTimeSlot(Function0<Unit> scrollToInstructions) {
    if (shouldScrollToCurrent) {
        scrollToInstructions.invoke();
    }
    shouldScrollToCurrent = true;
}
```

Also, I suggest to change the method name somewhat to let it become easier to understand from the calling side.

Reply:
Resolved in e47a60b6aca9109824935591c137bd820c8f1158.